src/application/common/logger.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import * as Logger from 'bunyan';
+
+const ILogger = Symbol("ILogger");
+
+interface ILogger {
+    addStream(stream: Logger.Stream): Promise<void>;
```

Comment:
1. Create child creates a new Bunyan Logger instance, so we would need to create a new set of LoggerClient/LoggerServer on our end to manage that...  Or find a way to map that...

2. Why do these have to be notifications ? 
This is a singleton interface so setting the level on the Logger once the promise is resolved would be ok no ?

3. OK I like that idea
4. OK nice